packages/compass-field-store/src/stores/store.ts

```diff
@@ -1,57 +1,14 @@
 import { createStore } from 'redux';
-import reducer, { changeFields } from '../modules';
-import type { Schema } from 'mongodb-schema';
-import parseSchema from 'mongodb-schema';
-import type { AppRegistry, ActivateHelpers } from 'hadron-app-registry';
-import { schemaFieldsToAutocompleteItems } from '../modules/fields';
+import reducer from '../modules';
+import type { ActivateHelpers } from 'hadron-app-registry';
 import { FieldStoreContext } from './context';
 
 export function activatePlugin(
   _initialProps: unknown,
-  { globalAppRegistry }: { globalAppRegistry: AppRegistry },
-  { on, cleanup }: ActivateHelpers
+  _: unknown,
+  { cleanup }: ActivateHelpers
 ) {
   const store = createStore(reducer);
 
-  const emitFieldsChanged = (ns: string) => {
-    const fieldsState = store.getState()[ns];
-    globalAppRegistry.emit('fields-changed', {
-      ns,
-      ...fieldsState,
-      autocompleteFields: schemaFieldsToAutocompleteItems(fieldsState.fields),
-    });
-  };
-
-  const onDocumentsChanged = async ({
-    ns,
-    docs,
-  }: {
-    ns: string;
-    docs: Document[];
-  }) => {
-    try {
-      const { fields } = await parseSchema(docs);
-      store.dispatch(changeFields(ns, fields));
-      emitFieldsChanged(ns);
-    } catch {
-      // ignore errors
-    }
-  };
-
-  on(globalAppRegistry, 'documents-refreshed', onDocumentsChanged);
-
-  on(globalAppRegistry, 'document-inserted', onDocumentsChanged);
-
-  on(globalAppRegistry, 'documents-paginated', onDocumentsChanged);
-
-  on(
-    globalAppRegistry,
-    'schema-analyzed',
-    ({ ns, schema }: { ns: string; schema: Schema }) => {
-      store.dispatch(changeFields(ns, schema.fields));
-      emitFieldsChanged(ns);
-    }
-  );
-
   return { store, deactivate: cleanup, context: FieldStoreContext };
```

Comment:
Now that this plugin is hoisted outside of the "connected" tree, would it make sense to set up something to clean up the store when disconnecting? Otherwise we might be accumulating too much data in memory that is not being used otherwise